core/transaction/validator.go

```diff
@@ -319,10 +320,17 @@ func CheckTransactionPayload(txn *Transaction) error {
 			return errors.New("asset is not enough")
 		}
 	case *payload.RegisterName:
-		name := pld.Name
-		registrant, err := Store.GetName(name)
+		name, err := Store.GetName(pld.Registrant)
+		if name != nil {
+			return errors.New(fmt.Sprintf("pubKey %+v already has registered name %s", pld.Registrant, *name))
```

Comment:
Shall we restrict to one name per pubkey? Personally I think it's reasonable to allow multiple name to be mapped to the same pubkey (alias), as long as each name is only mapped to one pubkey.